docs/source/gateway/index.rst

```diff
@@ -1,6 +1,1049 @@
 .. _gateway:
 
-MLflow Gateway
-==============
+================================
+MLflow AI Gateway (Experimental)
+================================
+
+.. warning::
+
+    The MLflow AI Gateway is a new, **experimental feature**. It is subject to modification, feature improvements, or feature removal without advance notice.
+
+The MLflow AI Gateway service is a powerful tool designed to streamline the usage and management of
+various language model providers, such as OpenAI and Anthropic, within an organization.
+It offers a high-level interface that simplifies the interaction with these services by providing
+a unified endpoint to handle all language model requests.
+
+A major advantage of using the MLflow AI Gateway service is its centralized management of API keys.
+By storing these keys in one secure location, organizations can significantly enhance their
+security posture by minimizing the exposure of sensitive API keys throughout the system. It also
+helps to prevent exposing these keys within code or requiring end-users to manage keys safely.
+
+The gateway is designed to be flexible and adaptable, capable of easily defining and managing routes by updating the
+configuration file. This enables the easy incorporation
+of new providers or provider model types into the system without necessitating changes to
+applications that interface with the gateway. This level of adaptability makes the MLflow AI Gateway
+Service an invaluable tool in environments that require agility and quick response to changes.
+
+This simplification and centralization of language model interactions, coupled with the added
+layer of security for API key management, make the MLflow AI Gateway service an ideal choice for
+organizations that use language models on a regular basis.
+
+.. contents:: Table of Contents
+  :local:
+  :depth: 3
+
+.. _gateway-concepts:
+
+Concepts
+========
+
+AI Gateway Configuration
+------------------------
+
+The MLflow AI Gateway service relies on a user-provided configuration file, written in YAML,
+that defines the routes and providers available to the service. The configuration file dictates
+how the gateway interacts with various language model providers and determines the end-points that
+users can access.
+
+.. note::
+
+    The configuration for the MLflow AI Gateway on `Databricks` is not performed with the use of a YAML configuration file. Rather, there are specific APIs that are used for management of the AI Gateway routes within `Databricks`.
+    Please consult the official `Databricks documentation <https://docs.databricks.com/machine-learning/index.html>`_ for guidance on configuring routes within `Databricks`.
```

Comment:
We won't be documenting these at https://docs.databricks.com/machine-learning/index.html for now (i.e. this isn't quite the link we want to use).

Instead, can we add Databricks-specific info for configuring gateway routes to the quickstart (https://github.com/mlflow/mlflow/pull/8763/files#r1253855943)?